packages/ember-htmlbars/lib/keywords/component.js

```diff
@@ -75,13 +106,23 @@ export default {
 };
 
 function render(morph, env, scope, params, hash, template, inverse, visitor) {
-  let componentPath = morph.state.componentPath;
+  const componentPath = morph.state.componentPath;
 
   // If the value passed to the {{component}} helper is undefined or null,
   // don't create a new ComponentNode.
   if (componentPath === undefined || componentPath === null) {
     return;
   }
-
-  env.hooks.component(morph, env, scope, componentPath, params, hash, { default: template, inverse }, visitor);
+  const result = lookupComponent(env.container, componentPath);
```

Comment:
I think you can avoid this temp variable via destructuring:

    const {
      component,
      layout
    } = lookupComponent(..........);